### ISSUE TITLE
SlangpyTraining Sample ignores selected loss

### ISSUE THREAD

**Issue by dom-wuest** [NVIDIA-RTX/RTXNS#8]
In the SlangpyTraining sample one can configure a loss, see [SlangpyTraining.py : 114](https://github.com/NVIDIA-RTX/RTXNS/blob/1a2513120190a43f61c23590811a8b80ec914bd7/samples/SlangpyTraining/SlangpyTraining.py#L114C5-L114C43), but this is not used in the shader. Instead, the TrainTexture and EvalLoss functions default to L2 loss:  [SlangpyTraining.slang : 119 / 142](https://github.com/NVIDIA-RTX/RTXNS/blob/1a2513120190a43f61c23590811a8b80ec914bd7/samples/SlangpyTraining/SlangpyTraining.slang#L142) 
```slang
float3 EvalLoss<Loss : rtxns::mlp::ILoss<float, 3>>(float2 inputUV, float3 predictedRGB, Texture2D<float4> targetTex)
{
    float3 targetRGB = SampleTexture(targetTex, inputUV).rgb;
    return rtxns::mlp::LossValueFromVector<3, rtxns::mlp::L2Relative<float, 3>>(
        targetRGB, predictedRGB, 1.0f);
}
```
the generic loss is already passed to this point, but its not used for calculation.

**Comment by dom-wuest** [comment]
Remotely related to this issue, I realized with the latest Release 1.3.1 you've added an epsilon to the loss terms. While it improves the convergence on the ShaderTraining example, it has a negative impact on the SlangpyTraining sample. This is likely due to the scale of the epsilon value, as in this example the loss is in the same order of magnitude as the epsilon. So the epsilon value should be adjustable. Either as a hyperparameter to the training or maybe add epsilon as a generic argument to the loss type?

**Comment by nv-aallan** [comment]
Thank you raising this! I have merged the PR.
Regarding an adjustable epsilon value, I have a change lined up. I will update next week.
